bridges/snowbridge/pallets/outbound-queue-v2/src/process_message_impl.rs

```diff
@@ -18,6 +18,11 @@ impl<T: Config> ProcessMessage for Pallet<T> {
 	) -> Result<bool, ProcessMessageError> {
 		let weight = T::WeightInfo::do_process_message();
 		if meter.try_consume(weight).is_err() {
+			Self::deposit_event(Event::MessageRejected {
+				id: None,
+				nonce: None,
+				error: ProcessMessageError::Overweight(weight),
+			});
```

Comment:
This won't be nessessary as the `Overweight` message will be retried in future blocks.

Reply:
It might still be useful to have visibility on when this happens though. If it is perpetually overweight it would be very hard to be debug without the event. See https://github.com/paritytech/polkadot-sdk/issues/7972

Reply:
Ron is right, the message is not rejected, but will be retried and might succeed. `MessageRejected` should signal the message was completely rejected, bridge will not process it.

Clara is right too that we should have visibility on this, otherwise it might be perpetually overweight and we have no way of seeing it.

I propose here you use a separate/dedicated event. Something like `Event::MessagePostponed { reason: overweight }` - and you could even use this postponed event when "Yield"ing because of max messages limit.